Back-end/index.js

```diff
@@ -225,3 +226,54 @@ app.post("/register", async (req, res) => {
     });
   }
 });
+
+app.get("/match", async (req, res) => {
+  let users = []; // Hold users when filtering by states
+  let matches = []; // Hold users when filtering by age
+  let results; // Similarity score
+  let map1 = new Map(); // Storing users with the similarity score as the key and the value using the user information
+  const { age, state, keyword } = req.body;
+  const data = await prisma.User.findMany({ // Gets users from the database
+    where: {
+      accountType: "Mentor", 
+    },
+  });
+  users.map((user) => {
+    if (user.state === state) {
+      users.push(user);
+    }
+  });
```

Comment:
Not sure if I missed anything: `users` is declared as an empty array in line 231, will line 242 be entered, if `users` is an empty array?

Reply:
i updated this in a small commit in main. I realized I made this typo and fixed it earlier today.